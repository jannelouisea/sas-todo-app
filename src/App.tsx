import React from 'react';
import './App.scss';
import {
    ToDoApp
} from 'src/components'
import { Provider } from 'react-redux';
import { store } from 'src/app/store';

function App() {
    return (
        <React.StrictMode>
            <Provider store={store}>
                <ToDoApp />
            </Provider>
        </React.StrictMode>
    );
}

export default App;
