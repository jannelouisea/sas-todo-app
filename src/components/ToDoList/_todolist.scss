.to-do-list {
    display: flex;
    flex-direction: column;
    overflow: hidden;
    overflow-y: auto;

    &_item {
        margin-bottom: 1rem;
    }

    &_msg {
        text-align: center;
        margin-top: 1rem;
    }
}