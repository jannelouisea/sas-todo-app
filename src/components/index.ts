import Item from 'src/components/Item/Item';

export {
    Item
}