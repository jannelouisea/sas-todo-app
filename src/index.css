@tailwind base;
@tailwind components;
@tailwind utilities;

html {
  background: url(static/img/milk.jpg) no-repeat center center fixed;
  -webkit-background-size: cover;
  -moz-background-size: cover;
  -o-background-size: cover;
  background-size: cover;
}

body {
  margin: 0;
  font-family: 'Public Sans', -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: 'Public Sans', source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
    monospace;
}

/* TODO - Move this to another css file */
/* Hide scrollbar for Chrome, Safari and Opera */
.no-scrollbar::-webkit-scrollbar {
  display: none;
}

/* Hide scrollbar for IE, Edge and Firefox */
.no-scrollbar {
  -ms-overflow-style: none;
  /* IE and Edge */
  scrollbar-width: none;
  /* Firefox */
}