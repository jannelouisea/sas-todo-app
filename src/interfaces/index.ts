import {
    Sort,
} from 'src/static/enums'

export type IToDoItem = {
    id: string,
    text: string,
    createdAt: Date,
    completed: boolean
}

export type ISortByItem = {
    value: Sort,
    label: string
}