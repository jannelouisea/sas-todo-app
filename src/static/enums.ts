export enum Sort {
    Incmpl1st = 'INCMPL_1ST',
    Cmpl1st = 'CMPL_1ST',
    DateAsc = 'DATE_ASC',
    DateDesc = 'DATE_DESC',
    TextAsc = 'TEXT_ASC',
    TextDesc = 'TEXT_DESC',
}

export enum MUIColor {
    Primary = 'primary',
    Success = 'success',
    Warning = 'warning',
    Neutral = 'neutral',
    Danger = 'danger',
}

export enum MUIIconColor {
    Inherit = 'inherit',
}

export enum MUIIconSize {
    Inherit = 'inherit',
}

export enum MUIVariant {
    Solid = 'solid',
    Soft = 'soft',
    Outlined = 'outlined',
    Plain = 'plain',
}

export enum MUISize {
    Small = 'sm',
    Medium = 'md',
    Large = 'lg',
}

export enum MUIPos {
    Right = 'right',
}

export enum MUIText {
    Label = 'body3'
}

export enum MUIFontSize {
    Small = 'small'
}