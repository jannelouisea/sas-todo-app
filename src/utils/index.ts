import isOlderItem from './isOlderItem';

export {
    isOlderItem,
}