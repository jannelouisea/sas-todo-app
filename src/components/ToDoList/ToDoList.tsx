import { useAppSelector } from 'src/app/hooks'
import { IToDoItem } from 'src/interfaces'
import { Item } from 'src/components'
import { isOlderItemUtil } from 'src/utils';

function ToDoList() {
    const NO_ITEMS_MATCH_MSG = 'No to-do items found that meet the current search and filter criteria.';
    const ITEMS_COMPLETED_MSG = 'Hooray! No items in your to-do list. ðŸŽ‰ (Are you sure you have nothing to-do?)';

    const todoItems = useAppSelector(state => state.todoItems);
    const todoItemsFilters = useAppSelector(state => state.todoItemsFilters);

    const todoItemsExist = todoItems.length > 0;

    const itemMeetsCriteria = (item: IToDoItem) => {
        const includesSearchTerm = todoItemsFilters.searchTerm === '' ? true : item.text.includes(todoItemsFilters.searchTerm);
        const isOldItem = isOlderItemUtil(item);
        return ((todoItemsFilters.showNewer && !item.completed && !isOldItem) ||
            (todoItemsFilters.showOlder && !item.completed && isOldItem) ||
            (todoItemsFilters.showCompleted && item.completed)) && includesSearchTerm;
    };

    const filteredItems = todoItems.filter(item => itemMeetsCriteria(item));
    const filteredItemsExist = filteredItems.length > 0;

    return (
        <div className="to-do-list no-scrollbar">
            {
                todoItemsExist ?
                    filteredItemsExist ?
                        filteredItems.map((item: IToDoItem) => <div key={item.id} className='to-do-list_item'><Item item={item} /></div>) :
                        <span className='to-do-list_msg'>{NO_ITEMS_MATCH_MSG}</span>
                    : <span className='to-do-list_msg'>{ITEMS_COMPLETED_MSG}</span>
            }
        </div>
    );
}

export default ToDoList;