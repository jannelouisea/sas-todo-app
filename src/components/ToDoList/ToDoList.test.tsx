import thunkMiddleware from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ToDoList from './ToDoList';
import { Sort } from 'src/static/enums';
import { isOlderItemUtil } from 'src/utils';
import moment from 'moment';

const middlewares = [thunkMiddleware];
const mockStore = configureMockStore(middlewares);

describe('ToDoList', () => {
    afterEach(() => {
        jest.clearAllMocks();
    });

    it('renders newer items correctly', () => {
        const state = {
            todoItems: [
                {
                    id: 'item-001',
                    text: 'Test newer',
                    createdAt: moment().toDate(),
                    completed: false
                },
                {
                    id: 'item-002',
                    text: 'Test newer again',
                    createdAt: moment().toDate(),
                    completed: false
                },
            ],
            todoItemsFilters: {
                searchTerm: '',
                showNewer: true,
                showOlder: true,
                showCompleted: true,
                sortBy: Sort.Incmpl1st,
            }
        };

        const { container } = render(
            <Provider store={mockStore(state)}>
                <ToDoList />
            </Provider>
        );

        // JLA
        // Checking actual container vs creating a snapshot since the newer dates will
        // always differ depending on when test was run.
        // Also, mocking isOlderItemUtils is not mocked at the Item component level
        const newerItems = container.getElementsByClassName('MuiCard-colorPrimary');
        const olderItems = container.getElementsByClassName('MuiCard-colorWarning');
        const completedItems = container.getElementsByClassName('MuiCard-colorSuccess');

        expect(newerItems.length).toBe(2);
        expect(olderItems.length).toBe(0);
        expect(completedItems.length).toBe(0);
    });

    it('renders older items correctly', () => {
        const state = {
            todoItems: [
                {
                    id: 'item-001',
                    text: 'Test older',
                    createdAt: moment('2023-07-18').toDate(),
                    completed: false
                },
                {
                    id: 'item-002',
                    text: 'Test older again',
                    createdAt: moment('2023-07-18').toDate(),
                    completed: false
                },
            ],
            todoItemsFilters: {
                searchTerm: '',
                showNewer: false,
                showOlder: true,
                showCompleted: false,
                sortBy: Sort.Incmpl1st,
            }
        };

        const tree = renderer
            .create(
                <Provider store={mockStore(state)}>
                    <ToDoList />
                </Provider>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders completed items correctly', () => {
        const state = {
            todoItems: [
                {
                    id: 'item-001',
                    text: 'Test completed',
                    createdAt: moment('2023-07-18').toDate(),
                    completed: true
                },
                {
                    id: 'item-002',
                    text: 'Test completed again',
                    createdAt: moment('2023-07-18').toDate(),
                    completed: true
                },
            ],
            todoItemsFilters: {
                searchTerm: '',
                showNewer: false,
                showOlder: false,
                showCompleted: true,
                sortBy: Sort.Incmpl1st,
            }
        };

        const tree = renderer
            .create(
                <Provider store={mockStore(state)}>
                    <ToDoList />
                </Provider>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    // Does not show newer items if newer filter is off
    it('does not render older items is showOlder filter is off', () => {
        const state = {
            todoItems: [
                {
                    id: 'item-001',
                    text: 'Test newer',
                    createdAt: moment().toDate(),
                    completed: false
                },
                {
                    id: 'item-002',
                    text: 'Test newer again',
                    createdAt: moment().toDate(),
                    completed: false
                },
            ],
            todoItemsFilters: {
                searchTerm: '',
                showNewer: false,
                showOlder: true,
                showCompleted: true,
                sortBy: Sort.Incmpl1st,
            }
        };

        const { container } = render(
            <Provider store={mockStore(state)}>
                <ToDoList />
            </Provider>
        );

        // JLA
        // Checking actual container vs creating a snapshot since the newer dates will
        // always differ depending on when test was run.
        // Also, mocking isOlderItemUtils is not mocked at the Item component level
        const newerItems = container.getElementsByClassName('MuiCard-colorPrimary');
        const olderItems = container.getElementsByClassName('MuiCard-colorWarning');
        const completedItems = container.getElementsByClassName('MuiCard-colorSuccess');

        expect(newerItems.length).toBe(0);
        expect(olderItems.length).toBe(0);
        expect(completedItems.length).toBe(0);
    });

    // Does not show newer items if older filter is off
    it('does not render older items is showOlder filter is off', () => {
        const state = {
            todoItems: [
                {
                    id: 'item-001',
                    text: 'Test older',
                    createdAt: moment('2023-07-18').toDate(),
                    completed: false
                },
                {
                    id: 'item-002',
                    text: 'Test older again',
                    createdAt: moment('2023-07-18').toDate(),
                    completed: false
                },
            ],
            todoItemsFilters: {
                searchTerm: '',
                showNewer: false,
                showOlder: false,
                showCompleted: false,
                sortBy: Sort.Incmpl1st,
            }
        };

        const tree = renderer
            .create(
                <Provider store={mockStore(state)}>
                    <ToDoList />
                </Provider>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('does not render completed items is showCompleted filter is off', () => {
        const state = {
            todoItems: [
                {
                    id: 'item-001',
                    text: 'Test completed',
                    createdAt: moment('2023-07-18').toDate(),
                    completed: true
                },
                {
                    id: 'item-002',
                    text: 'Test completed again',
                    createdAt: moment('2023-07-18').toDate(),
                    completed: true
                },
            ],
            todoItemsFilters: {
                searchTerm: '',
                showNewer: false,
                showOlder: false,
                showCompleted: false,
                sortBy: Sort.Incmpl1st,
            }
        };

        const tree = renderer
            .create(
                <Provider store={mockStore(state)}>
                    <ToDoList />
                </Provider>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders items that match search term', () => {
        const state = {
            todoItems: [
                {
                    id: 'item-001',
                    text: 'testing',
                    createdAt: moment('2023-07-18').toDate(),
                    completed: true
                },
                {
                    id: 'item-002',
                    text: 'not matching',
                    createdAt: moment('2023-07-18').toDate(),
                    completed: true
                },
            ],
            todoItemsFilters: {
                searchTerm: 'testing',
                showNewer: true,
                showOlder: true,
                showCompleted: true,
                sortBy: Sort.Incmpl1st,
            }
        };

        const tree = renderer
            .create(
                <Provider store={mockStore(state)}>
                    <ToDoList />
                </Provider>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    // Renders msg when no items exist
    it('renders msg when no to-do items exist', () => {
        const state = {
            todoItems: [],
            todoItemsFilters: {
                searchTerm: '',
                showNewer: true,
                showOlder: true,
                showCompleted: true,
                sortBy: Sort.Incmpl1st,
            }
        };

        const tree = renderer
            .create(
                <Provider store={mockStore(state)}>
                    <ToDoList />
                </Provider>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
