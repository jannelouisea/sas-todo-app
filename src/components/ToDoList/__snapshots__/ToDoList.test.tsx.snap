// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoList does not render completed items is showCompleted filter is off 1`] = `
<div
  className="to-do-list no-scrollbar"
>
  <span
    className="to-do-list_msg"
  >
    No to-do items found that meet the current search and filter criteria.
  </span>
</div>
`;

exports[`ToDoList does not render older items is showOlder filter is off 1`] = `
<div
  className="to-do-list no-scrollbar"
>
  <span
    className="to-do-list_msg"
  >
    No to-do items found that meet the current search and filter criteria.
  </span>
</div>
`;

exports[`ToDoList renders completed items correctly 1`] = `
<div
  className="to-do-list no-scrollbar"
>
  <div
    className="to-do-list_item"
  >
    <div
      className="MuiCard-root MuiCard-vertical MuiCard-variantSoft MuiCard-colorSuccess MuiCard-sizeSm drop-shadow css-17iutky-JoyCard-root"
    >
      <div
        className="to-do-item"
        data-first-child=""
        data-last-child=""
      >
        <div
          className="to-do-item_toggle"
        >
          <button
            aria-label="Complete item button"
            className="MuiIconButton-root MuiIconButton-variantSolid MuiIconButton-colorSuccess MuiIconButton-sizeSm css-qxb87h-JoyIconButton-root"
            id="complete-item-button"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="DoneIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M9 16.2 4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
              />
            </svg>
          </button>
        </div>
        <div
          className="to-do-item_text"
        >
          <span>
            Test completed
          </span>
          <div
            className="item-created-date"
          >
            <p
              className="MuiTypography-root MuiTypography-body3 MuiTypography-colorNeutral MuiTypography-variantPlain css-1oc79or-JoyTypography-root"
            >
              Created Jul 18, 2023 12:00 AM
            </p>
            <div
              className="hidden"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-17e890r-MuiSvgIcon-root"
                data-testid="HourglassBottomIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m18 22-.01-6L14 12l3.99-4.01L18 2H6v6l4 4-4 3.99V22h12zM8 7.5V4h8v3.5l-4 4-4-4z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="to-do-item_remove-btn"
        >
          <button
            aria-label="Remove item button"
            aria-labelledby={null}
            className="MuiIconButton-root MuiIconButton-variantPlain MuiIconButton-colorSuccess MuiIconButton-sizeSm css-1raploq-JoyIconButton-root"
            id="remove-item-button"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1ekxpud-MuiSvgIcon-root"
              data-testid="DeleteIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="to-do-list_item"
  >
    <div
      className="MuiCard-root MuiCard-vertical MuiCard-variantSoft MuiCard-colorSuccess MuiCard-sizeSm drop-shadow css-17iutky-JoyCard-root"
    >
      <div
        className="to-do-item"
        data-first-child=""
        data-last-child=""
      >
        <div
          className="to-do-item_toggle"
        >
          <button
            aria-label="Complete item button"
            className="MuiIconButton-root MuiIconButton-variantSolid MuiIconButton-colorSuccess MuiIconButton-sizeSm css-qxb87h-JoyIconButton-root"
            id="complete-item-button"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="DoneIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M9 16.2 4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
              />
            </svg>
          </button>
        </div>
        <div
          className="to-do-item_text"
        >
          <span>
            Test completed again
          </span>
          <div
            className="item-created-date"
          >
            <p
              className="MuiTypography-root MuiTypography-body3 MuiTypography-colorNeutral MuiTypography-variantPlain css-1oc79or-JoyTypography-root"
            >
              Created Jul 18, 2023 12:00 AM
            </p>
            <div
              className="hidden"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-17e890r-MuiSvgIcon-root"
                data-testid="HourglassBottomIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m18 22-.01-6L14 12l3.99-4.01L18 2H6v6l4 4-4 3.99V22h12zM8 7.5V4h8v3.5l-4 4-4-4z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="to-do-item_remove-btn"
        >
          <button
            aria-label="Remove item button"
            aria-labelledby={null}
            className="MuiIconButton-root MuiIconButton-variantPlain MuiIconButton-colorSuccess MuiIconButton-sizeSm css-1raploq-JoyIconButton-root"
            id="remove-item-button"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1ekxpud-MuiSvgIcon-root"
              data-testid="DeleteIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ToDoList renders items that match search term 1`] = `
<div
  className="to-do-list no-scrollbar"
>
  <div
    className="to-do-list_item"
  >
    <div
      className="MuiCard-root MuiCard-vertical MuiCard-variantSoft MuiCard-colorSuccess MuiCard-sizeSm drop-shadow css-17iutky-JoyCard-root"
    >
      <div
        className="to-do-item"
        data-first-child=""
        data-last-child=""
      >
        <div
          className="to-do-item_toggle"
        >
          <button
            aria-label="Complete item button"
            className="MuiIconButton-root MuiIconButton-variantSolid MuiIconButton-colorSuccess MuiIconButton-sizeSm css-qxb87h-JoyIconButton-root"
            id="complete-item-button"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="DoneIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M9 16.2 4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
              />
            </svg>
          </button>
        </div>
        <div
          className="to-do-item_text"
        >
          <span>
            testing
          </span>
          <div
            className="item-created-date"
          >
            <p
              className="MuiTypography-root MuiTypography-body3 MuiTypography-colorNeutral MuiTypography-variantPlain css-1oc79or-JoyTypography-root"
            >
              Created Jul 18, 2023 12:00 AM
            </p>
            <div
              className="hidden"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-17e890r-MuiSvgIcon-root"
                data-testid="HourglassBottomIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m18 22-.01-6L14 12l3.99-4.01L18 2H6v6l4 4-4 3.99V22h12zM8 7.5V4h8v3.5l-4 4-4-4z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="to-do-item_remove-btn"
        >
          <button
            aria-label="Remove item button"
            aria-labelledby={null}
            className="MuiIconButton-root MuiIconButton-variantPlain MuiIconButton-colorSuccess MuiIconButton-sizeSm css-1raploq-JoyIconButton-root"
            id="remove-item-button"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1ekxpud-MuiSvgIcon-root"
              data-testid="DeleteIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ToDoList renders msg when no to-do items exist 1`] = `
<div
  className="to-do-list no-scrollbar"
>
  <span
    className="to-do-list_msg"
  >
    Hooray! No items in your to-do list. 🎉 (Are you sure you have nothing to-do?)
  </span>
</div>
`;

exports[`ToDoList renders older items correctly 1`] = `
<div
  className="to-do-list no-scrollbar"
>
  <div
    className="to-do-list_item"
  >
    <div
      className="MuiCard-root MuiCard-vertical MuiCard-variantOutlined MuiCard-colorWarning MuiCard-sizeSm drop-shadow css-1l37zdt-JoyCard-root"
    >
      <div
        className="to-do-item"
        data-first-child=""
        data-last-child=""
      >
        <div
          className="to-do-item_toggle"
        >
          <button
            aria-label="Complete item button"
            className="MuiIconButton-root MuiIconButton-variantOutlined MuiIconButton-colorWarning MuiIconButton-sizeSm css-pxh0lg-JoyIconButton-root"
            id="complete-item-button"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="DoneIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M9 16.2 4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
              />
            </svg>
          </button>
        </div>
        <div
          className="to-do-item_text"
        >
          <span>
            Test older
          </span>
          <div
            className="item-created-date"
          >
            <p
              className="MuiTypography-root MuiTypography-body3 MuiTypography-colorNeutral MuiTypography-variantPlain css-gfqtt1-JoyTypography-root"
            >
              Created Jul 18, 2023 12:00 AM
            </p>
            <div
              className=""
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-17e890r-MuiSvgIcon-root"
                data-testid="HourglassBottomIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m18 22-.01-6L14 12l3.99-4.01L18 2H6v6l4 4-4 3.99V22h12zM8 7.5V4h8v3.5l-4 4-4-4z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="to-do-item_remove-btn"
        >
          <button
            aria-label="Remove item button"
            aria-labelledby={null}
            className="MuiIconButton-root MuiIconButton-variantPlain MuiIconButton-colorWarning MuiIconButton-sizeSm css-punmis-JoyIconButton-root"
            id="remove-item-button"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1ekxpud-MuiSvgIcon-root"
              data-testid="DeleteIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="to-do-list_item"
  >
    <div
      className="MuiCard-root MuiCard-vertical MuiCard-variantOutlined MuiCard-colorWarning MuiCard-sizeSm drop-shadow css-1l37zdt-JoyCard-root"
    >
      <div
        className="to-do-item"
        data-first-child=""
        data-last-child=""
      >
        <div
          className="to-do-item_toggle"
        >
          <button
            aria-label="Complete item button"
            className="MuiIconButton-root MuiIconButton-variantOutlined MuiIconButton-colorWarning MuiIconButton-sizeSm css-pxh0lg-JoyIconButton-root"
            id="complete-item-button"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="DoneIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M9 16.2 4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
              />
            </svg>
          </button>
        </div>
        <div
          className="to-do-item_text"
        >
          <span>
            Test older again
          </span>
          <div
            className="item-created-date"
          >
            <p
              className="MuiTypography-root MuiTypography-body3 MuiTypography-colorNeutral MuiTypography-variantPlain css-gfqtt1-JoyTypography-root"
            >
              Created Jul 18, 2023 12:00 AM
            </p>
            <div
              className=""
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-17e890r-MuiSvgIcon-root"
                data-testid="HourglassBottomIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m18 22-.01-6L14 12l3.99-4.01L18 2H6v6l4 4-4 3.99V22h12zM8 7.5V4h8v3.5l-4 4-4-4z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="to-do-item_remove-btn"
        >
          <button
            aria-label="Remove item button"
            aria-labelledby={null}
            className="MuiIconButton-root MuiIconButton-variantPlain MuiIconButton-colorWarning MuiIconButton-sizeSm css-punmis-JoyIconButton-root"
            id="remove-item-button"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1ekxpud-MuiSvgIcon-root"
              data-testid="DeleteIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
